internal/security/bootstrapper/command/genpassword/command_test.go

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package genpassword
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/interfaces"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+func TestNewCommand(t *testing.T) {
+	// Arrange
+	ctx := context.Background()
+	wg := &sync.WaitGroup{}
+	lc := logger.MockLogger{}
+	config := &config.ConfigurationStruct{}
+
+	tests := []struct {
+		name        string
+		cmdArgs     []string
+		expectedErr bool
+	}{
+		{"Good: genPasswd cmd empty option", []string{}, false},
+		{"Bad: genPasswd invalid option", []string{"--invalid=xxx"}, true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			command, err := NewCommand(ctx, wg, lc, config, tt.cmdArgs)
+			if tt.expectedErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.NotNil(t, command)
+			}
+		})
+	}
+}
+
+func TestExecute(t *testing.T) {
+	// Arrange
+	ctx := context.Background()
+	wg := &sync.WaitGroup{}
+	lc := logger.MockLogger{}
+	config := &config.ConfigurationStruct{}
+
+	genPwd, err := NewCommand(ctx, wg, lc, config, []string{})
+	require.NoError(t, err)
+	require.NotNil(t, genPwd)
+	require.Equal(t, "genPassword", genPwd.GetCommandName())
+
+	statusCode, err := genPwd.Execute()
```

Comment:
Any way to capture STDOUT to verify a password was generated?

Reply:
i don't know anyway to capture that feasibly. it could be dangerous if we can do that so that password would be leaking easily.